### ISSUE TITLE
Any chance to support golang's template syntax?

### ISSUE THREAD

**Issue by davyzhang** [Glavin001/atom-beautify#243]
Here's the doc: http://golang.org/pkg/text/template/

if there's no hope to support this

can add a function using regex match to skip some text from format?

**Comment by Glavin001** [comment]
Will have to investigate this further to see I can support this in Atom Beautify or not.

Could you provide me with a simple example, sample input and expected beautified output? Like a hello world that I could write tests for, etc. Thanks.

**Comment by davyzhang** [comment]
```
function nav(dir) {
  var postData = {
    game_id: "{{.ActiveGame.ID}}",
    src: $("#txtSrc").val(),
    trans: $("#txtTrans").val(),
    untrans: $("#untranslated").is(":checked"),
    direction: dir,
    {{if eq .Mode 0 }} //note this should not be valid js/html syntax
    image: VisualTranslation.getPic(),
    overlay: VisualTranslation.getData(),
    {{end}}
    tid: trans_id,
    lang:"{{.ActiveLang}}",
    mode:"{{.Mode}}"
  };

  $.ajax({
    url: "/api/next",
    type: "POST",
    data: postData
    )}
  }
```
if I formatted it will turns out to be like this
```
function nav(dir) {
  var postData = {
    game_id: "{{.ActiveGame.ID}}",
    src: $("#txtSrc").val(),
    trans: $("#txtTrans").val(),
    untrans: $("#untranslated").is(":checked"),
    direction: dir,
    {
      {
        if eq.Mode 0 //check this block out
      }
    }
    image: VisualTranslation.getPic(),
    overlay: VisualTranslation.getData(),
    {
      {
        end //and this block
      }
    }
    tid: trans_id,
    lang: "{{.ActiveLang}}",
    mode: "{{.Mode}}"
  };

  $.ajax({
      url: "/api/next",
      type: "POST",
      data: postData
    )
  }
}

```
so I think just leave everything in {{.*}} unformated is fine


Plus:
Atom Beautify amazingly works well on golang html template,
it well chane
```
{{$actGame := .ActiveGame}} 
{{$actLang := .ActiveLang}} 
{{range $key,$game := .Games}}
```
to
```
{{$actGame := .ActiveGame}} {{$actLang := .ActiveLang}} {{range $key,$game := .Games}}
```
but that's ok

here's the formatted html template, looks nice.

```
{{define "nav"}}
<nav class="navbar navbar-default navbar-static-top" style="margin-bottom: 0;min-height:0">
  <div class="navbar-default sidebar">
    <div class="sidebar-nav navbar-collapse">
      <ul id="side-menu" class="nav collapse in">
        {{$actGame := .ActiveGame}} {{$actLang := .ActiveLang}} {{range $key,$game := .Games}} //here
        <li><a href="#"><i class="fa fa-gamepad fa-fw"></i> {{$game.Display}}<span class="fa arrow"></span></a>
          <ul class="nav nav-second-level collapse {{if $actGame}} {{if eq $actGame.ID $game.ID }}in{{end}}{{end}}">
            <li><a href="/tool?game_id={{$game.ID}}"><i class="fa fa-wrench fa-fw"></i> tool</a>
            </li>
            {{range $idx,$lang := $game.Langs}}
            <li><a href="#"><i class="fa fa-language fa-fw"></i> {{$lang.ToDisplay}}<span class="fa arrow"></span></a>
              <ul class="nav nav-third-level collapse {{if eq $actLang $lang}}in{{end}}">
                <li><a href="/client?lang={{$lang}}&game_id={{$game.ID}}&mode=0">design </a>
                </li>
                <li><a href="/client?lang={{$lang}}&game_id={{$game.ID}}&mode=1">write </a>
                </li>
                <li><a href="/client?lang={{$lang}}&game_id={{$game.ID}}&mode=2">review</a>
                </li>
            </li>
            </ul>
        </li>
        {{end}}
        </ul>
        </li>
        {{end}}
        <li>
          <a href="/admin"><i class="fa fa-gears fa-fw"></i> 管理</a>
        </li>
      </ul>
    </div>
  </div>
</nav>
{{end}}
```

**Comment by Glavin001** [comment]
@prettydiff, is this similar to any languages you currently support?

**Comment by prettydiff** [comment]
It seems from the example code that there is a sample that resembles JavaScript and a sample that resembles HTML. Which of these formats should be representational of a golang template?

https://github.com/prettydiff/prettydiff/issues/84

**Comment by Glavin001** [comment]
@prettydiff does this sample pass already? Maybe all that is left is to add a language to Atom Beautify?

**Comment by prettydiff** [comment]
I have forgotten about this.  I will need to add extra support for the `{{if }}` convent to the JS Beautifier, so this is not ready yet.

**Comment by prettydiff** [comment]
Support published to Pretty Diff v1.16.29.  You can reinstall Atom Beautify or update its NPM dependencies to get the latest.
